experimental/devices/mfrc522/mfrc522.go

```diff
@@ -0,0 +1,757 @@
+	copy(data[6:], accessData[:4])
+	copy(data[10:], keyB[:])
+	return r.write(calcBlockAddress(sector&0xFF, 3), data)
+}
+
+// Auth authenticate the card fof the sector/block using the provided data.
+//
+// 	mode - the authentication mode.
+// 	sector - the sector to authenticate on.
+// 	block - the block within sector to authenticate.
+// 	sectorKey - the key to be used for accessing the sector data.
+// 	serial - the serial of the card.
+func (r *Dev) Auth(mode byte, sector, block int, sectorKey []byte, serial []byte) (AuthStatus, error) {
+	return r.auth(mode, calcBlockAddress(sector, block), sectorKey, serial)
+}
+
+// ReadCard reads the card sector/block.
+//
+// 	auth - the authentication mode.
+// 	sector - the sector to authenticate on.
+// 	block - the block within sector to authenticate.
+// 	key - the key to be used for accessing the sector data.
+func (r *Dev) ReadCard(auth byte, sector int, block int, key []byte) (data []byte, err error) {
+	defer func() {
+		if err == nil {
+			err = r.StopCrypto()
+		}
+	}()
+	uuid, err := r.selectCard()
+	if err != nil {
+		return
+	}
+	state, err := r.Auth(auth, sector, block, key, uuid)
+	if err != nil {
+		return
+	}
+	if state != AuthOk {
+		err = wrapf("can not authenticate")
+		return
+	}
+	return r.ReadBlock(sector, block)
+}
+
+// ReadAuth - read the card authentication data.
+//
+// 	sector - the sector to authenticate on.
+// 	key - the key to be used for accessing the sector data.
+func (r *Dev) ReadAuth(auth byte, sector int, key []byte) (data []byte, err error) {
+	defer func() {
+		if err == nil {
+			err = r.StopCrypto()
+		}
+	}()
+	uuid, err := r.selectCard()
+	if err != nil {
+		return
+	}
+	state, err := r.Auth(auth, sector, 3, key, uuid)
+	if err != nil {
+		return
+	}
+	if state != AuthOk {
+		return nil, wrapf("can not authenticate")
+	}
+
+	return r.read(calcBlockAddress(sector, 3))
+}
+
+// CalculateBlockAccess calculates the block access.
+func CalculateBlockAccess(ba *BlocksAccess) []byte {
+	res := make([]byte, 4)
+	res[0] = (^ba.getBits(1) & 0x0F) | ((^ba.getBits(2) & 0x0F) << 4)
+	res[1] = (^ba.getBits(3) & 0x0F) | (ba.getBits(1) & 0x0F << 4)
+	res[2] = (ba.getBits(2) & 0x0F) | (ba.getBits(3) & 0x0F << 4)
+	res[3] = res[0] ^ res[1] ^ res[2]
+	return res
+}
+
+// ParseBlockAccess parses the given byte array into the block access structure.
+func ParseBlockAccess(ad []byte) *BlocksAccess {
+	ba := new(BlocksAccess)
+	ba.B0 = BlockAccess(ad[1]&0x10>>4 | ad[2]&0x01<<1 | ad[2]&0x10>>2)
+	ba.B1 = BlockAccess(ad[1]&0x20>>5 | ad[2]&0x02 | ad[2]&0x20>>3)
+	ba.B2 = BlockAccess(ad[1]&0x40>>6 | ad[2]&0x04>>1 | ad[2]&0x40>>4)
+	ba.B3 = SectorTrailerAccess(ad[1]&0x80>>7 | ad[2]&0x08>>2 | ad[2]&0x80>>5)
+	return ba
+}
+
+func (r *Dev) String() string {
+	return fmt.Sprintf("Mifare MFRC522 [bus: %v, reset pin: %s, irq pin: %s]",
+		r.spiDev, r.resetPin.Name(), r.irqPin.Name())
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Not a fan of these, I agree something like:

```
// Dev private/helper functions
```

is helpful to categorize the code, but no need to spend 5 lines on this.